docker/entrypoint.sh

```diff
@@ -15,16 +15,19 @@ fi
 
 OLYMPIA_USER="olympia"
 
+function get_olympia_uid() { echo "$(id -u "$OLYMPIA_USER")"; }
+function get_olympia_gid() { echo "$(id -g "$OLYMPIA_USER")"; }
+
 if [[ -n "${HOST_UID:-}" ]]; then
   usermod -u ${HOST_UID} ${OLYMPIA_USER}
   echo "${OLYMPIA_USER} UID: ${OLYMPIA_UID} -> ${HOST_UID}"
-fi
 
-uid=$(id -u $OLYMPIA_USER)
-gid=$(id -g $OLYMPIA_USER)
+  # Ensure the olympia user has access to the /deps directory
+  time chown -R "$(get_olympia_uid):$(get_olympia_gid)" /deps
```

Comment:
As a follow-up to speed things up (*): If we stop mounting /deps maybe we could skip this and install deps as `root` when building the image - or any user really, including the uid=9500 - as long as the right `umask` is set at install to allow reading and executing by others.

The only benefits of having the `olympia` user actually owning those files are:
- Consistency with prod behavior (where deps are installed with `olympia` user with uid 9500 atm)
- Ability to edit the files in deps when entering the container as the `olympia` user

(*) I don't think we want to block the PR on this. Not sure what the implications would be for users of the `docker-compose.deps.yml` file. Plus, now that `/deps` is purely internal, this seems to be relatively fast.

Reply:
I've tried, haven't found a solution, but there could be one eventually.

Currently the chown takes around 30-45 seconds. Not catastrophically long, faster than the update_deps did. But yes, it would be nice to remove the need for this, we just have to figure out how to enable olympia user access to host mounted files and container created files that would have different owners. 

So far, when I tried to share ownership via group, pip and npm complained about permissions but happy to see a PR if you have a solution in mind.